### ISSUE TITLE
Disable language server for syntax test files by default

### ISSUE THREAD

**Issue by jfcherng** [sublimelsp/LSP#2504]
## TL;DR

By default, disable servers for

- syntax test files
- SublimeREPL views

## Description

I think enabling servers on syntax test files only cause (linting) issues. Only very few ones (e.g., LSP-copilot) will work well. [By suggested rules](https://www.sublimetext.com/docs/syntax.html#testing), the name of a syntax test file starts with `syntax_test_` (but in `Default/run_syntax_tests.py`, it finds filenames start with `syntax_test`).

The same reason for disabling for SublimeREPL views. See https://github.com/sublimelsp/LSP-pyright/issues/343
A SublimeREPL view has the view setting `"repl": True,`.

## Suggestion

Something equivalent to

```py
    @classmethod
    def should_ignore(cls, view: sublime.View) -> bool:
        return bool(
            # SublimeREPL views
            view.settings().get("repl")
            # syntax test files
            or os.path.basename(view.file_name() or "").startswith("syntax_test")
        )
```

**Comment by jwortmann** [comment]
I'm not sure whether `view.settings().get("repl")` would be a good approach; LSP currently registers new (non-file) views in `on_activated_async`. Dependent on how/when exactly SublimeREPL writes this setting, this approach might be prone to a race condition, i.e. whether this setting is set or read first.

In general, if we want to have this, I would probably prefer a more general setting name like for example `lsp_ignore` which would not be specific to SublimeREPL.
